sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -120,6 +121,17 @@ func main() {
 	}
 }
 
+// locateModule resolves a node module name to a file path that can be loaded
+func locateModule(mod string, nodePath string) (string, error) {
+	program := fmt.Sprintf("console.log(require.resolve('%s'));", mod)
```

Comment:
I am not sure if we care about this but this makes me a little uneasy from a security perspective, since `PULUMI_LANGUAGE_NODEJS_RUN_PATH` is user-controlled. It seems bad to me that you can slap javascript in `PULUMI_LANGUAGE_NODEJS_RUN_PATH` and have it get run.

I'm not really sure how to fix this so I leave this up to you about whether or not you care - just seemed a little off to me.

Reply:
Do we even need to support `PULUMI_LANGUAGE_NODEJS_RUN_PATH` anymore, given that with Matt's change you can now simply arrange to have NPM's module resolution resolve to a different version, if you'd like to do so?

Reply:
Hmm, that is probably true. The only place I know of that we use this is in the nodejs language host tests and we can hack `NODE_PATH` for that purpose, I think.

Reply:
> The only place I know of that we use this is in the nodejs language host tests and we can hack NODE_PATH for that purpose, I think.

I'm not 100% sure how this would work, given the fact that we don't have a layout that is compatible with the normal node module structure.  From the test where we use this, we have this giant comment:

```
    // Normally, `pulumi-language-nodejs` launches `./node-modules/@pulumi/pulumi/cmd/run` which is responsible
    // for setting up some state and then running the actual user program.  However, in this case, we don't
    // have a folder structure like the above because we are seting the package as we've built it, not it installed
    // in another application.
    //
    // `pulumi-language-nodejs` allows us to set `PULUMI_LANGUAGE_NODEJS_RUN_PATH` in the environment, and when
    // set, it will use that path instead of the default value. For our tests here, we set it and point at the
    // just built version of run.
```

And we set it to `./bin/cmd/run`  Even with `NODE_PATH` hacking, I think we'd still need to have something that looked like `node_modules/@pulumi/pulumi` on disk, which we don't have today.  You are correct that we only use this feature in our tests today.

I'll also note that we are no worse with this change than we were before, since we allowed `PULUMI_LANGUAGE_NODEJS_RUN_PATH` before this change.  We could instead consider adding something like -runPath to `pulumi-language-nodejs` so instead of an environment variable it is on the command line.  I'm not sure if you think that makes things less "user-controlled" or not.  Note that at the end of the day, `pulumi-language-nodejs` launches `run.js` which `requires()` whatever you passed on the command line as your project root, so if you can run `pulumi-language-nodejs` and your goal was to run some arbitrary javascript, I think you already won.